### DISCUSSION TITLE
How do maintainers feel about refactoring?

### DISCUSSION THREAD

**Discussion by julie777** [pyscaffold/pyscaffold#597] [Dev Discussions]
(Belongs in the developer/development/maintenance/maintainer category if we have one.)

While working on a solution for #594 I noticed that some of the code related to git (repo.py) is mixed in with the shell code (shell.py).

How do maintainers feel about refactoring things like this?

(I know that I am new. I'm trying to ask questions to understand the group dynamic so that I can fit in. I really don't want to step on any toes or get lost on a tangent.)

**Comment by FlorianWilhelm** [comment]
My suggestion would be to open up an issue about this, then within that issue one could lay out an alternative approach and how it would be refactored. This will lead to a small discussion, potentially followed then by a PR with the consent of this discussion. 

That being said, @abravalheri has been the much more active developer in the recent years and much of the code in these modules is from here. So I think it would be nice to have an exchange with him.

**Comment by julie777** [comment]
@FlorianWilhelm I thought this was starting a discussion about the topic. Do we always need an issue to have a discussion?
Or, it the issue just a way to get everyone's attention?

@abravalheri please add your thoughts.

**Reply by FlorianWilhelm** [reply]
Sorry, you are right, a discussion is more or less the same. I am using the Discussions feature of Github still not enough and rather think in terms of issues.

**Reply by julie777** [reply]
I understand. When I started using GitHub it didn't have all the nice features like discussions, wiki, actions, etc.

**Comment by abravalheri** [comment]
Hi @julie777, I think refactoring is always welcome and a nice way to nourish the code base so it survive for a long time!

The suggestion from @FlorianWilhelm is a very nice way of doing it!

**Comment by julie777** [comment]
I just wanted to get a little feedback before I spent too much time on something. 

I am a big fan of automated tests and every time you touch the code leave it a little better than you found it.
